source/fundamentals/builders/filters.txt

```diff
@@ -11,3 +11,560 @@ Filters Builders
+   {"_id": 6, "color": "pink", "qty": 5, "vendor": ["C"]}
+
+
+.. _bitwise:
+
+Bitwise
+-------
+
+The bitwise operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Bitwise Method
+     - Matches
+      
+   * - ``bitsAllSet()``
+     - documents when the specified bits of a field are set (i.e. 1).
+
+   * - ``bitsAllClear()``
+     - documents when the specified bits of a field are clear (i.e. 0).
+      
+   * - ``bitsAnySet()``
+     - documents when at least one of the specified bits of a field are set (i.e. 1).
+     
+   * - ``bitsAnyClear()``
+     - documents when at least one of the specified bits of a field are clear (i.e. 0).
+
+The following example matches documents that have a bitField field
+with bits set at positions of the corresponding bitmask 50 (i.e.
+00110010) in the ``binary_numbers`` collection: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Filters.java
+   :language: java
+   :dedent:
+   :start-after: begin bitsComparison
+   :end-before: end bitsComparison
+
+The BSON renders as:
+
+.. code-block:: json
+      :copyable: false
+   
+   { 
+      "a": 
+         { 
+            "$bitsAllSet": [ 1, 5 ] 
+         } 
+   }
+
+The output of the above code snippet should look something like this:
+   
+.. code-block:: json
+   :copyable: false
+      
+   { _id: 9, a: 54, binaryValue: "00110110" }
+   { _id: 12, a: 102, binaryValue: "01100110" }
+
+.. _geospatial:
+
+Geospatial
+----------
+
+The geospatial operator methods include:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Geospatial Method 
+     - Matches
+      
+   * - ``geoWithin()``
+     - documents containing a GeoJSON geometry field that falls within within a bounding GeoJSON geometry.
+
+   * - ``geoWithinBox()``
+     - documents containing a grid coordinates field that exist within the specified box.
+      
+   * - ``geoWithinPolygon()``
+     - documents containing a grid coordinates field that exist within the specified polygon.
+
+   * - ``geoWithinCenter()``
+     - documents containing a grid coordinates field that exist within the specified circle.
+
+   * - ``geoWithinCenterSphere()``
+     - geometries containing a geospatial data field (GeoJSON or legacy coordinate pairs) that exist within the specified circle, using spherical geometry.
+
+   * - ``geoIntersects()``
+     - geometries that intersect with a GeoJSON geometry. The 2dsphere index supports $geoIntersects.
+
+   * - ``near()``
+     - geospatial objects in proximity to a point. Requires a geospatial index. The 2dsphere and 2d indexes support $near.
+
+   * - ``nearSphere()``
+     - geospatial objects in proximity to a point on a sphere. Requires a geospatial index. The 2dsphere and 2d indexes support $nearSphere.
+
+
+The following example creates a filter that matches documents in which
+the ``point`` field contains a GeoJSON Geometry object that falls within
```

Comment:
Question: should "Geometry" be capitalized here (I'm not sure)? And is geometry redundant since you specify GeoJSON, or is there a geometry type of GeoJSON?

Reply:
Just checked in the docs manual and the way it's used there is "GeoJSON geometry"